src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx

```diff
@@ -210,30 +211,33 @@ function BarplotViz(props: VisualizationProps) {
   );
 
   // prettier-ignore
-  // allow 2nd parameter of resetCheckedLegendItems for checking legend status
   const onChangeHandlerFactory = useCallback(
-    < ValueType,>(key: keyof BarplotConfig, resetCheckedLegendItems?: boolean) => (newValue?: ValueType) => {
-      const newPartialConfig = resetCheckedLegendItems
-        ? {
-            [key]: newValue,
-            checkedLegendItems: undefined
-          }
-        : {
-          [key]: newValue
-        };
-       updateVizConfig(newPartialConfig);
+    < ValueType,>(key: keyof BarplotConfig, resetCheckedLegendItems?: boolean, resetAxisRanges?: boolean) => (newValue?: ValueType) => {
+      const newPartialConfig = {
+        [key]: newValue,
+        ...(resetCheckedLegendItems ? { checkedLegendItems: undefined } : {}),
+	...(resetAxisRanges ? { dependentAxisRange: undefined } : {}),
+      };
+      updateVizConfig(newPartialConfig);
+      if (resetAxisRanges)
+	setTruncatedDependentAxisWarning('');
```

Comment:
Sorry the formatting is terrible.  The `prettier-ignore` is/was needed for the `< ValueType,>` bit (I don't remember why - I think maybe it upsets the code colouring in my emacs...).  We could try removing the `prettier-ignore` directive.